File: arrays & hashmaps/valid-anagram/valid-anagram-1.js
var isAnagram = function(s, t) {
    if(s.length !== t.length){
        return false
    }
    let countS = {}
    let countT = {}

    for(let i = 0;i < s.length; i++){
        if(countS.hasOwnProperty(s[i])){
            countS[s[i]]++
        } else {
            countS[s[i]] = 1;
        }
        if(countT.hasOwnProperty(s[i])){
            countT[s[i]]++
        } else {
            countT[s[i]] = 1;
        }
    }
    for(let j in countS){
        if(countS[j] !== countT[j]){
            return false;
        }
    } 
    return true;
};

File: arrays & hashmaps/valid-anagram/valid-anagram-1.py
def isAnagram(s, t): 
    if len(s) != len(t):                        #if lengths are different, cannot be anagram
        return False                            #return false

    countS, countT = {}, {}                     #initialize two dictionaries

    for i in range(len(s)):                     #for i in range of length of an anagram
        countS[s[i]] = 1 + countS.get(s[i], 0)  #dict[value(letter in anagram s)] = number_of_occurances
        countT[t[i]] = 1 + countT.get(t[i], 0)  #dict[value(letter in anagram t)] = number_of_occurances
    
    for c in countS:                            #for c in length of hashDict
        if countS[c] != countT.get(c, 0):       #if value for key of countS != value for key of countT
            return False                        #return False
        
    return True                #no problems, return True, are anagrams

#51 ms
#15.3 MB

print(isAnagram("anagram", "nagaram"))
# output : True